code/modules/projectiles/gun.dm

```diff
@@ -166,12 +169,14 @@
 			step(user, user.inertia_dir)
 
 	if(silenced)
-		playsound(user, fire_sound, 10, 1)
+		if(fire_sound)
+			playsound(user, fire_sound, 10, 1)
 	else
-		playsound(user, fire_sound, 50, 1)
-		user.visible_message("<span class='warning'>[user] fires [src][reflex ? " by reflex":""]!</span>", \
-		"<span class='warning'>You fire [src][reflex ? "by reflex":""]!</span>", \
-		"You hear a [istype(in_chamber, /obj/item/projectile/beam) ? "laser blast" : "gunshot"]!")
+		if(fire_sound)
+			playsound(user, fire_sound, 50, 1)
+			user.visible_message("<span class='warning'>[user] fires [src][reflex ? " by reflex":""]!</span>", \
```

Comment:
Objectively, if a gun has no fire sound then one could argue that people wouldn't hear it happen via the message.
Subjectively, I put that under the same if() because the blast cannon has its own fire message that doesn't match the format of the stock gun fire message at all, so this way it won't be displayed for the blast cannon because it has no fire_sound.